proposals/dao/balancer_gauge_fix.ts

```diff
@@ -0,0 +1,116 @@
+  const balancerGaugeStakerProxy = await ProxyFactory.deploy(
+    balancerGaugeStakerImplementation.address,
+    addresses.proxyAdmin,
+    '0x' // empty calldata
+  );
+  await balancerGaugeStakerProxy.deployTransaction.wait();
+  const balancerGaugeStaker = await ethers.getContractAt('BalancerGaugeStaker', balancerGaugeStakerProxy.address);
+  logging && console.log('balancerGaugeStaker: ', balancerGaugeStaker.address);
+
+  return {
+    balancerGaugeStakerImplementation,
+    balancerGaugeStaker
+  };
+};
+
+// Do any setup necessary for running the test.
+// This could include setting up Hardhat to impersonate accounts,
+// ensuring contracts have a specific state, etc.
+const setup: SetupUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  console.log(`Setup of ${fipNumber} : reading CR oracle...`);
+
+  // make sure TC has enough ETH to run
+  await forceEth(addresses.tribalCouncilSafe);
+  await forceEth(addresses.tribalCouncilTimelock);
+
+  // make sure ETH oracle is fresh (for B.AMM not to revert, etc)
+  // Read Chainlink ETHUSD price & override chainlink storage to make it a fresh value
+  const ethPrice = (await contracts.chainlinkEthUsdOracleWrapper.read())[0].toString() / 1e10;
+  await overwriteChainlinkAggregator(addresses.chainlinkEthUsdOracle, Math.round(ethPrice), '8');
+
+  // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+  // assume non-zero B-70WETH-30FEI were unstaked & moved to the PCVDeposit
+  // before this proposal's execution.
+  // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+  const lpTokenHolder = '0x4f9463405f5bc7b4c1304222c1df76efbd81a407';
+  const lpTokenSigner = await getImpersonatedSigner(lpTokenHolder);
+  await forceEth(lpTokenHolder);
+  await contracts.balancerFeiWethPool.connect(lpTokenSigner).transfer(addresses.balancerDepositFeiWeth, `1000${e18}`);
+
+  // read pcvStats before proposal execution
+  pcvStatsBefore = await contracts.collateralizationOracle.pcvStats();
+};
+
+// Tears down any changes made in setup() that need to be
+// cleaned up before doing any validation checks.
+const teardown: TeardownUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  console.log(`No actions to complete in teardown for fip${fipNumber}`);
+};
+
+// Run any validations required on the fip using mocha or console logging
+// IE check balances, check state of contracts, etc.
+const validate: ValidateUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  // check the new staking contract has staked in gauge
+
+  // check rewards can be claimed
+  await time.increase('3600');
+  console.log('BAL balance before claim', (await contracts.bal.balanceOf(addresses.balancerGaugeStaker)) / 1e18);
+  await contracts.balancerGaugeStaker.mint(addresses.balancerFeiWethPool);
+  console.log('BAL balance after claim ', (await contracts.bal.balanceOf(addresses.balancerGaugeStaker)) / 1e18);
```

Comment:
Being pedantic, but worthwhile adding an explicit check that balanceBefore != balanceAfter?

Reply:
yes for sure ! still in dev though and I usually start with prints ^^